src/wallet/gtest/test_rpc_wallet.cpp

```diff
@@ -284,12 +284,14 @@ TEST(WalletRPCTests, RPCZsendmanyTaddrToSapling)
     pwalletMain->LoadWalletTx(wtx);
 
     // Context that z_sendmany requires
-    auto builder = TransactionBuilder(consensusParams, nextBlockHeight, std::nullopt, pwalletMain);
+    auto builder = WalletTxBuilder(Params(), *pwalletMain, minRelayTxFee);
     mtx = CreateNewContextualCMutableTransaction(consensusParams, nextBlockHeight, false);
 
-    auto selector = pwalletMain->ZTXOSelectorForAddress(taddr, true, false).value();
-    std::vector<ResolvedPayment> recipients = { ResolvedPayment(std::nullopt, pa, 1*COIN, Memo::FromHexOrThrow("ABCD")) };
-    TransactionStrategy strategy(PrivacyPolicy::AllowRevealedSenders);
+    auto selector = pwalletMain->ZTXOSelectorForAddress(taddr, true, false, TransactionStrategy(PrivacyPolicy::FullPrivacy)).value();
+    std::vector<Payment> recipients = { Payment(pa, 1*COIN, Memo::FromHexOrThrow("ABCD")) };
+    // we need AllowFullyTransparent because the transaction will result
+    // in transparent change as a consequence of sending from a legacy taddr
+    TransactionStrategy strategy(PrivacyPolicy::AllowFullyTransparent);
```

Comment:
Similarly doesn't make sense. At a minimum we should be using a consistent strategy here:
```suggestion
    // we need AllowFullyTransparent because the transaction will result
    // in transparent change as a consequence of sending from a legacy taddr
    TransactionStrategy strategy(PrivacyPolicy::AllowFullyTransparent);
    auto selector = pwalletMain->ZTXOSelectorForAddress(taddr, true, false, strategy).value();
    std::vector<Payment> recipients = { Payment(pa, 1*COIN, Memo::FromHexOrThrow("ABCD")) };
```

Reply:
Hrmm, I thought I corrected this already. Maybe I missed this instance, or maybe it was in one of the later PRs. In any case, it’s applied now. Thanks.